state/modelgeneration.go

```diff
@@ -171,6 +171,55 @@ func assignGenerationUnitTxnOps(id, appName, unitName string) []txn.Op {
 	}
 }
 
+// Complete marks the generation as completed.
+func (g *Generation) Complete() error {
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if attempt > 0 {
+			if err := g.Refresh(); err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+		if !g.Active() {
+			return nil, errors.New("next generation is not currently active")
+		}
+		ok, err := g.CanAutoComplete()
```

Comment:
If we choose another name for `Complete` we should change this name too, along with the error prose below.

Reply:
done